### ISSUE TITLE
Create Colab notebook with `denoising_diffusion_pytorch` example

### ISSUE THREAD

**Issue by sgbaird** [sparks-baird/xtal2png#57]
https://www.assemblyai.com/blog/diffusion-models-for-machine-learning-introduction/

**Comment by sgbaird** [comment]
Not working on Colab https://github.com/lucidrains/denoising-diffusion-pytorch/issues/33

**Comment by sgbaird** [comment]
Working on this as a script for now: https://github.com/sparks-baird/xtal2png/blob/main/notebooks/ddpm.py

Has been producing less favorable results #79 #80. The priority is to get bookends attached #12 so that hyperparameter optimization can take place on several models.

**Comment by vinz97** [comment]
Hi, i'm working con Colab too. How can i print inception score (IS), FID score and NLL (Negative Log Likelihoods) during the training for comparing them to the values written in the article?

**Comment by sgbaird** [comment]
@vinz97 by the article do you mean the [blog post](https://www.assemblyai.com/blog/diffusion-models-for-machine-learning-introduction/)? I'd suggest asking this as an issue on https://github.com/lucidrains/denoising-diffusion-pytorch. I briefly perused the code and previous issues and didn't find the answer. As a side note, for FID you might consider using https://github.com/GaParmar/clean-fid. For inception score, you might consider https://github.com/sbarratt/inception-score-pytorch, though [they recommend against using it as a performance metric for generative models](https://arxiv.org/abs/1801.01973). For NLL, 
[learned_gaussian_diffusion.py#L119-L120](https://github.com/lucidrains/denoising-diffusion-pytorch/blob/cf6db719854c3180ff122b9d557c89c4b796cc37/denoising_diffusion_pytorch/learned_gaussian_diffusion.py#L119-L120) is probably relevant.

You might also consider using [`imagen-pytorch`](https://github.com/lucidrains/imagen-pytorch) in image-to-image mode.

**Comment by sgbaird** [comment]
@vinz97 btw I answered assuming you weren't asking about anything specific to `xtal2png`. If that wasn't the case, please let me know.

**Comment by vinz97** [comment]
@sgbaird Yeah sorry for the confusing question, i was asking about the code [https://github.com/lucidrains/denoising-diffusion-pytorch](https://github.com/lucidrains/denoising-diffusion-pytorch. ).
I'm using the code written in the README file of the repository (under the USAGE paragraph, the second one). My aim is to training the CIFAR10 dataset and trying to print the Inception score, FID and NLL for comparing them to the values written [in this article](https://arxiv.org/pdf/2006.11239.pdfl).
But i can use only Colab, so the training is really slowly, especially for 700k steps

**Comment by sgbaird** [comment]
See https://github.com/sparks-baird/xtal2png/blob/main/scripts/denoising_diffusion_pytorch_example.py and https://github.com/sparks-baird/xtal2png/blob/main/scripts/denoising_diffusion_pytorch_pretrained_sample.py instead
